servers/src/test/java/tachyon/master/EditLogTest.java

```diff
@@ -0,0 +1,242 @@
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.master;
+
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.codec.binary.Base64;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.type.TypeReference;
+
+import tachyon.TachyonURI;
+import tachyon.conf.TachyonConf;
+
+/**
+ * Unit Test for EditLog. Test the read/write correctness of each operation.
+ */
+public class EditLogTest {
+  private EditLog mEditLog = null;
+  private TachyonConf mTachyonConf = new TachyonConf();
+  private String mTmpPath = null;
+
+  @Test
+  public void addBlockTest() throws IOException {
+    // Write ADD_BLOCK
+    mEditLog.addBlock(1, 2, 128L, 1409349750338L);
+    mEditLog.flush();
+    mEditLog.close();
+
+    // Read ADD_BLOCK and check
+    EditLogOperation op = getSingleOpFromFile(mTmpPath);
+    Assert.assertEquals(101, op.mTransId);
+    Assert.assertEquals(EditLogOperationType.ADD_BLOCK, op.mType);
+    Assert.assertEquals(1, op.getInt("fileId").intValue());
+    Assert.assertEquals(2, op.getInt("blockIndex").intValue());
+    Assert.assertEquals(128L, op.getLong("blockLength").longValue());
+    Assert.assertEquals(1409349750338L, op.getLong("opTimeMs").longValue());
+  }
+
+  @Test
+  public void addCheckpointTest() throws IOException {
+    // Write ADD_CHECKPOINT
+    TachyonURI checkpointPath = new TachyonURI("/test/checkpoint");
+    mEditLog.addCheckpoint(1, 256L, checkpointPath, 1409349750338L);
+    mEditLog.flush();
+    mEditLog.close();
+
+    // Read ADD_CHECKPOINT and check
+    EditLogOperation op = getSingleOpFromFile(mTmpPath);
+    Assert.assertEquals(101, op.mTransId);
+    Assert.assertEquals(EditLogOperationType.ADD_CHECKPOINT, op.mType);
+    Assert.assertEquals(1, op.getInt("fileId").intValue());
+    Assert.assertEquals(256L, op.getLong("length").longValue());
+    Assert.assertEquals(checkpointPath.toString(), op.getString("path"));
+    Assert.assertEquals(1409349750338L, op.getLong("opTimeMs").longValue());
+  }
+
+  @After
```

Comment:
I'll move them to the front.
BTW, I see some other unit test organize the methods in alpha-beta order (e.g., the JournalIntegrationTest).